### PR TITLE
Added type hints for TF: TransfoXL

### PR DESCRIPTION
**Pull request by thliang01**
Based on Issue #16059 
I have added type hints for Tensorflow TransfoXL Model.

### CODE REVIEW

File: src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py [hunk 1]
```diff
@@ -542,14 +542,15 @@ def _update_mems(self, hids, mems, mlen, qlen):
...
-        training=False,
+        input_ids: Optional[TFModelInputType] = None,
+        mems: Optional[List[tf.Tensor]] = None,
+        head_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        inputs_embeds: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        training: Optional[bool] = False,
```
Comment (Rocketknight1) [inline src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py:553]:
```suggestion
        training: bool = False,
```

File: src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py [hunk 2]
```diff
@@ -894,14 +895,14 @@ def __init__(self, config, *inputs, **kwargs):
...
-        return_dict=None,
-        training=False,
+        input_ids: Optional[TFModelInputType] = None,
+        mems: Optional[List[tf.Tensor]] = None,
+        head_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        inputs_embeds: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        training: Optional[bool] = False,
```
Comment (Rocketknight1) [inline src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py:905]:
```suggestion
        training: bool = False,
```

File: src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py [hunk 3]
```diff
@@ -974,15 +975,15 @@ def init_mems(self, bsz):
...
-        training=False,
+        input_ids: Optional[TFModelInputType] = None,
+        mems: Optional[List[tf.Tensor]] = None,
+        head_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        inputs_embeds: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        labels: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        training: Optional[bool] = False,
```
Comment (Rocketknight1) [inline src/transformers/models/transfo_xl/modeling_tf_transfo_xl.py:986]:
```suggestion
        training: bool = False,
```

**Review by Rocketknight1 — APPROVED** [review]
LGTM! One minor nitpick is that `training` should not be `None`, and so it shouldn't be marked as `Optional`, but other than that this is perfect, so thank you!

**Comment by thliang01** [comment]
I removed these Optional types as your suggested.
Holp you may check it and then merge my request.
@Rocketknight1

**Comment by Rocketknight1** [comment]
Looks good to me, thank you!

**Pull request merged**